influxql/engine.go

```diff
@@ -496,11 +496,19 @@ func (m *MapReduceJob) processAggregate(c *Call, reduceFunc ReduceFunc, resultVa
 		}
 	}
 
+	firstInterval := m.interval
+	if !m.stmt.IsRawQuery {
```

Comment:
How much unnecessary work does it avoid? 

Assuming we're not planning to write a test (if we are cool), this is a little bit concerning. We've got a new line of code, in a known problematic area, that computes the first interval in a bucket under certain conditions. Whether it's wrapped in a condition or not has no effect on whether the test suite passes. This makes me nervous. We'll be in the code again, and might break it.

I'll get off my soapbox.